OptimizelySDK/Bucketing/DecisionService.cs

```diff
@@ -253,5 +246,156 @@ public void SaveVariation(Experiment experiment, Variation variation, UserProfil
+        /// Get the variation if the user is bucketed for one of the experiments on this feature flag.
+        /// </summary>
+        /// <param name = "featureFlag" >The feature flag the user wants to access.</param>
+        /// <param name = "userId" >User Identifier</param>
+        /// <param name = "filteredAttributes" >The user's attributes. This should be filtered to just attributes in the Datafile.</param>
+        /// <returns>null if the user is not bucketed into the rollout or if the feature flag was not attached to a rollout.
+        /// otherwise the Variation the user is bucketed into</returns>
+        public virtual Variation GetVariationForFeatureExperiment(FeatureFlag featureFlag, string userId, UserAttributes filteredAttributes)
+        {
+            if (featureFlag.ExperimentIds == null || featureFlag.ExperimentIds.Count == 0)
+            {
+                Logger.Log(LogLevel.INFO, $"The feature flag \"{featureFlag.Key}\" is not used in any experiments.");
+                return null;
+            }
+
+            foreach (var experimentId in featureFlag.ExperimentIds)
+            {
+                var experiment = ProjectConfig.GetExperimentFromId(experimentId);
+
+                if (string.IsNullOrEmpty(experiment.Key))
+                    continue;
+
+                var variation = GetVariation(experiment, userId, filteredAttributes);
+
+                if (variation != null)
+                {
+                    Logger.Log(LogLevel.INFO, $"The user \"{userId}\" is bucketed into experiment \"{experiment.Key}\" of feature \"{featureFlag.Key}\".");
+                    return variation;
+                }
+            }
+
+            Logger.Log(LogLevel.INFO, $"The user \"{userId}\" is not bucketed into any of the experiments on the feature \"{featureFlag.Key}\".");
+            return null;
+        }
+
+        /// <summary>
+        /// Get the variation the user is bucketed into for the FeatureFlag
+        /// </summary>
+        /// <param name = "featureFlag" >The feature flag the user wants to access.</param>
+        /// <param name = "userId" >User Identifier</param>
+        /// <param name = "filteredAttributes" >The user's attributes. This should be filtered to just attributes in the Datafile.</param>
+        /// <returns>null if the user is not bucketed into any variation or the Variation the user is bucketed into
+        /// if the user is successfully bucketed.</returns>
+        public Variation GetVariationForFeature(FeatureFlag featureFlag, string userId, UserAttributes filteredAttributes)
+        {
+            // Check if the feature flag has an experiment and the user is bucketed into that experiment.
+            var variation = GetVariationForFeatureExperiment(featureFlag, userId, filteredAttributes);
+
+            if (variation != null)
+                return variation;
+
+            // Check if the feature flag has rollout and the the user is bucketed into one of its rules.
```

Comment:
good abstraction in this method. It makes it a bit easier to understand the order of operations 👍